### PR TITLE
Added cabal-fmt to pre-commit-hooks and reformatted all cabal files

### PR DESCRIPTION
**Pull request by zeme-iohk**
* PLT-42 Added cabal-fmt to pre-commit hooks & reformatted all existing cabal files
* Created fix-cabal-fmt script and added to shell
* Added CI test for cabal-fmt
* Updated CONTRIBUTING.adoc to include addition of cabal-fmt
* Replaced occurences of iohk/plutus with input-output-hk/plutus-apps

### CODE REVIEW

File: nix/tests/cabal-fmt.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  cd ..
+  diff --brief --recursive orig cabal > /dev/null
+  EXIT_CODE=$?
+  if [[ $EXIT_CODE != 0 ]]
+  then
+    mkdir -p $out/nix-support
+    diff -ur orig cabal > $out/cabal.diff
+    echo "file none $out/cabal.diff" > $out/nix-support/hydra-build-products
+    echo "*** cabal-fmt found changes that need addressed first"
+    echo "*** Please run \`fix-stylish-haskell\` and commit changes"
```
Comment (michaelpj) [inline nix/tests/cabal-fmt.nix:33]: Wrong tool. Maybe wrong in plutus-apps too.

File: nix/tests/cabal-fmt.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ runCommand, fixCabalFmt, src, lib, diffutils, glibcLocales }:
+let
+  src' = lib.cleanSourceWith {
```
Comment (michaelpj) [inline nix/tests/cabal-fmt.nix:3]: What about `lib.sourceFilesBySuffix src [ ".cabal" ]`?

**Review by michaelpj — APPROVED** [review]

**Comment by michaelpj** [comment]
> Updated CONTRIBUTING.adoc to include addition of cabal-fmt

Missing?

File: plutus-core/plutus-core.cabal [hunk 3]
```diff
@@ -676,85 +682,85 @@ executable generate-cost-model
...
+  main-is:          Main.hs
+  build-depends:
+    , base             >=4.9     && <5
+    , criterion        >=1.5.9.0
+    , index-envs
+    , nonempty-vector
+    , ral              ==0.1
+    , random           >=1.2.0
 
+-- broken for ral-0.2 conflicts with cardano-binary:recursion-schemes
```
Comment (sjoerdvisscher) [inline plutus-core/plutus-core.cabal:751]: Misplaced comment

File: plutus-metatheory/plutus-metatheory.cabal [hunk 4]
```diff
@@ -1,63 +1,78 @@
...
-  -- This makes cabal rebuild if any of these files change, which allow the
-  -- custom setup to fire and rebuild the Haskell sources
-  src/**/*.lagda,
+  Plutus.agda-lib
+  README.md
+  src/**/*.lagda
   src/**/*.lagda.md
-build-type:          Custom
+
+-- This makes cabal rebuild if any of these files change, which allow the
```
Comment (sjoerdvisscher) [inline plutus-metatheory/plutus-metatheory.cabal:20]: Misplaced comment

File: plutus-benchmark/plutus-benchmark.cabal [hunk 5]
```diff
@@ -1,301 +1,292 @@
...
-
+  import:         lang
+  type:           exitcode-stdio-1.0
+  main-is:        Spec.hs
+  hs-source-dirs: nofib/test
+  build-depends:
+    , base                                            >=4.9 && <5
+    , nofib-internal
+    , plutus-benchmark-common
+    , plutus-core:{plutus-core, plutus-core-testlib}  ^>=1.0
```
Comment (sjoerdvisscher) [inline plutus-benchmark/plutus-benchmark.cabal:143]: 🤯 TIL!

**Review by sjoerdvisscher — APPROVED** [review]

File: nix/tests/cabal-fmt.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ runCommand, fixCabalFmt, src, lib, diffutils, glibcLocales }:
+let
+  src' = lib.sourceFilesBySuffix src [ ".cabal" ];
```
Comment (michaelpj) [inline nix/tests/cabal-fmt.nix:3]: This is why you need CI and to test things yourself and not trust me :D It's `sourceFilesBySuffices`

**Comment by michaelpj** [comment]
Sorry, I edited one of the cabal files so now this has conflicts.

**Comment by michaelpj** [comment]
I'm also about to edit one again, so maybe hold off an hour or two.

**Comment by michaelpj** [comment]
Okay, I'm doing messing with cabal files, go!

**Comment by michaelpj** [comment]
You need to rerun the formatter, it's not happy.

**Comment by michaelpj** [comment]
Hmm, one of the packages isn't building properly, another bug in the formatter?

**Pull request merged**